### DISCUSSION TITLE
Avalonia MVVM binding TextBox.Text to getter-only property doesn't update text when getter output changes

### DISCUSSION THREAD

**Discussion by yesseruser** [AvaloniaUI/Avalonia#16573] [Q&A] [answered]
I have a calculator app using MVVM. The TextBox that shows the current number is read-only, and its Text is bound to a property (using Binding) that returns the current number stored in another variable as string with some additional logic (adding comma/dot at the end if it is supposed to be added before next number). 

When the return value of this property changes (e.g. I change the underlying number variable), the TextBox doesn't update. Why? Do I need to call some method to update the UI?

I have checked and yes, the logic of changing the underlying variable runs.

**Comment by thevortexcloud** [comment, marked as answer]
Normally in MVVM, your view model raises a change notification when the property changes (typically in the property setter). However with a get only property you need to manually and directly raise the change event. Specifically though, you need to implement `INotifyPropertyChanged` and raise the event that it implements.

https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged?view=net-8.0

**Reply by yesseruser** [reply]
Thank you! The ObserveableObject class which my ViewModel inherits has an already implemented OnPropertyChanged method.

**Comment by tobyfirth** [comment]
If the logic for recalculating value is triggered by changes to other properties you could use `NotifyPropertyChangedFor` attribute to trigger a property change notification for your read-only property. This explains this it a little more [Notifying dependent properties](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#notifying-dependent-properties).
